src/test/java/hudson/plugins/deploy/glassfish/GlassFish3xAdapterTest.java

```diff
@@ -103,44 +100,44 @@ public void testConfigureRemote() throws IOException, InterruptedException, Exec
      * This test only runs in your local environment
      * @throws IOException
      * @throws InterruptedException
-     */
-    //@Test
+     *
+    @Test
     public void testDeploy() throws IOException, InterruptedException {
-        
         adapter.redeploy(new FilePath(new File("src/test/simple.war")), "contextPath", null, null, new StreamBuildListener(System.out));
     }
     
-    //@Test
+    @Test
     public void testRemoteDeploy() throws IOException, InterruptedException {
-       
-
         remoteAdapter.redeploy(new FilePath(new File("src/test/simple.war")), "contextPath", null, null, new StreamBuildListener(System.out));
-    }
+    }*/
     
     @Test
-    public void testVariables() throws IOException, InterruptedException, ExecutionException {
+    public void testVariables() throws Exception {
+        Node n = jenkinsRule.createSlave();
     	EnvironmentVariablesNodeProperty property = new EnvironmentVariablesNodeProperty();
+
     	EnvVars envVars = property.getEnvVars();
     	envVars.put(homeVariable, home);
     	envVars.put(usernameVariable, username);
     	envVars.put(adminPortVariable, adminPort);
     	envVars.put(hostnameVariable, hostname);
     	jenkinsRule.jenkins.getGlobalNodeProperties().add(property);
 
-        FreeStyleProject project = jenkinsRule.createFreeStyleProject();
+        FreeStyleProject project = jenkinsRule.getInstance().createProject(FreeStyleProject.class, "fsp");
+        project.setAssignedNode(n);
         FreeStyleBuild build = project.scheduleBuild2(0).get();
         BuildListener listener = new StreamBuildListener(new ByteArrayOutputStream());
 
         adapter = new  GlassFish3xAdapter(getVariable(homeVariable), password, getVariable(usernameVariable), getVariable(adminPortVariable), null);
         Configuration config = new DefaultConfigurationFactory().createConfiguration(adapter.getContainerId(), ContainerType.REMOTE, ConfigurationType.RUNTIME);
-        adapter.configure(config, build.getEnvironment(listener), build.getBuildVariableResolver());
+        adapter.configure(config, project.getEnvironment(n, listener), build.getBuildVariableResolver());
         
         Assert.assertEquals(username, config.getPropertyValue(RemotePropertySet.USERNAME));
         Assert.assertEquals(adminPort, config.getPropertyValue(GlassFishPropertySet.ADMIN_PORT));
         
         remoteAdapter = new  GlassFish3xAdapter(null, password, getVariable(usernameVariable), getVariable(adminPortVariable), getVariable(hostnameVariable));
         config = new DefaultConfigurationFactory().createConfiguration(adapter.getContainerId(), ContainerType.REMOTE, ConfigurationType.RUNTIME);
-        remoteAdapter.configure(config, build.getEnvironment(listener), build.getBuildVariableResolver());
+        remoteAdapter.configure(config, project.getEnvironment(n, listener), build.getBuildVariableResolver());
```

Comment:
Manually tested variable expansion with Tomcat 8 and `TC_USER`+`TC_PASSWORD` set in the global Jenkins config. By debugging I was able to figure out that `build.getEnvironment()` gets variables only unique to the build, which global variables are not.

I think this test is sufficient. A more correct way to test this would be with a running container and building a project with a `DeployPublisher` added. IMO out of the scope of this PR.